### ISSUE TITLE
train and val loss -------Overfitting？

### ISSUE THREAD

**Issue by leemengxing** [xingyizhou/CenterNet#499]
I trained 5 specific categories of dlav0 version of centernet on coco, and loaded the centernet model of dlav0-80 you gave, but eventually the loss on the training set can drop to about 0.5, but the loss of val is about 3.What is the problem？Any help is greatly appreciated

**Comment by leemengxing** [comment]
python main.py ctdet --exp_id coco_dlav0_34 --batch_siz│
e 64 --master_batch 9 --arch dlav0_34 --lr 2.5e-4 --gpus 0,1,2,3 --num_workers 16 │
--num_epochs 230  --lr_step 180,210  --load_model ../models/ctdet_coco_dlav0_1x.pt│
h

**Comment by leemengxing** [comment]
![{75CC79C2-4579-4145-ABBE-67DA847115F0}_20191202110118](https://user-images.githubusercontent.com/24696984/69927499-26df5380-14f3-11ea-965f-29f23c4d525d.jpg)
![{DD0D0ADA-1580-4673-BF3E-4E515231357D}_20191202105934](https://user-images.githubusercontent.com/24696984/69927500-28108080-14f3-11ea-8b24-bf60e7348a0b.jpg)

**Comment by zhiqwang** [comment]
A question here, what's the mAP of the last checkpoints on the validation dataset? And there are a similar question in #34 .

**Comment by leemengxing** [comment]
The result is correct?I don't quite understand. As training progresses, val-loss gradually increases. Any explanation?

**Comment by zhiqwang** [comment]
I'm not sure, maybe the mAP can say something? I only train centernet on a small dataset, the training and validation loss behaviour are not consistent, but val-loss decrease in my case.

**Comment by leemengxing** [comment]
I tried testing MAP on the val dataset and I don't think this is a good result. I extracted coco5 data, the train data size is 12111, and the val size is 512. Is this inappropriate?

**Comment by leemengxing** [comment]
Maybe I should try using pascol's dataset hyperparameters

**Comment by zhiqwang** [comment]
Try decreasing the learning rate?

**Comment by leemengxing** [comment]
![{7F552DCB-DA3E-4D71-994E-A257E9673450}_20191204114836](https://user-images.githubusercontent.com/24696984/70111064-0d254400-168c-11ea-8cbf-5a691babbc4c.jpg)

**Comment by zhiqwang** [comment]
The magnitude of dataset size doesn't matter with the behaviour of loss curve, I think. Make sure the hyperparameter of your datasets is consistent with your setup? Maybe I am wrong.

**Comment by leemengxing** [comment]
I extracted specific 5 categories from coco, not the complete coco dataset. So hyperparameters may need to be adjusted

**Comment by leemengxing** [comment]
![{13814145-C1D3-4B9E-8700-C3787D22ED92}_20191206110647](https://user-images.githubusercontent.com/24696984/70292216-89e52900-1818-11ea-87d3-cdac1c531855.jpg)

**Comment by leemengxing** [comment]
Can I look at val-hm visually, neither my val-wh nor val-off loss will cause such problems.

**Comment by marscep456** [comment]
@leemengxing  @xingyizhou I ran into the same problem:The training loss continuous descending convergence,But the val loss rise.By looking at the val_loss log find the hm loss continuous increase,The other two losses continued to fall? 
Could you give me some suggest? @leemengxing  @xingyizhou  and thank you very much!

**Comment by ZJC1013** [comment]
> @leemengxing @xingyizhou I ran into the same problem:The training loss continuous descending convergence,But the val loss rise.By looking at the val_loss log find the hm loss continuous increase,The other two losses continued to fall?
> Could you give me some suggest? @leemengxing @xingyizhou and thank you very much!

I see the same when training on KITTI that val_hm_loss keeps increasing while all the losses in train_loss decrease nicely. Any idea on why this is happening?

**Comment by czc00125** [comment]
> 
> 
> python main.py ctdet --exp_id coco_dlav0_34 --batch_siz│
> e 64 --master_batch 9 --arch dlav0_34 --lr 2.5e-4 --gpus 0,1,2,3 --num_workers 16 │
> --num_epochs 230 --lr_step 180,210 --load_model ../models/ctdet_coco_dlav0_1x.pt│
> h

Hello, How could you to visualize the loss curve during training, canyou give me some instructions for this

**Comment by czc00125** [comment]
Hello, How could you to visualize the loss curve during training, can you give me some instructions for this,thank you !

**Comment by EphChem** [comment]
> > @leemengxing @xingyizhou I ran into the same problem:The training loss continuous descending convergence,But the val loss rise.By looking at the val_loss log find the hm loss continuous increase,The other two losses continued to fall?
> > Could you give me some suggest? @leemengxing @xingyizhou and thank you very much!
> 
> I see the same when training on KITTI that val_hm_loss keeps increasing while all the losses in train_loss decrease nicely. Any idea on why this is happening?

@ZJC1013 , nice to see you're working on this!  Have you resolved the issue using kitti data?

**Comment by ZJC1013** [comment]
@EphChem Wow, what a small research community we are in! Nice to see you here! So regarding this issue, I am also seeing that the validation hm_loss results in a U-shape. However, as suggested by author @xingyizhou in [Issue 148](https://github.com/xingyizhou/CenterNet/issues/148), this is somewhat not overfitting and unrelated to the AP detection performance, which I can also confirm with my own experiments on KITTI.

**Comment by EphChem** [comment]
> @EphChem Wow, what a small research community we are in! Nice to see you here! So regarding this issue, I am also seeing that the validation hm_loss results in a U-shape. However, as suggested by author @xingyizhou in [Issue 148](https://github.com/xingyizhou/CenterNet/issues/148), this is somewhat not overfitting and unrelated to the AP detection performance, which I can also confirm with my own experiments on KITTI.

Awesome!  Thanks @ZJC1013 !  Yes, small world indeed!  Hope to catch up with you soon.
